arrays/arrays.c

```diff
@@ -47,13 +58,21 @@ void destroy_array(Array *arr) {
 void resize_array(Array *arr) {
 
   // Create a new element storage with double capacity
+  int doubleCapacity = arr->capacity * 2;
+  char **newElements = realloc(arr->elements, doubleCapacity * sizeof(char *));
```

Comment:
Don't think you need realloc() and to copy each element over in a for loop. One or the other should suffice.